### PR TITLE
minetime: init at 1.4.12

### PR DESCRIPTION
**Pull request by dtzWill**
###### Motivation for this change

Builds on #54693, making small changes for adding more attributes.

Open to suggestions for better approach, but this works well for me :).

###### Things done

- [x] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nox --run "nox-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Assured whether relevant documentation is up to date
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

---

### CODE REVIEW

File: pkgs/build-support/appimage/default.nix [hunk 1]
```diff
@@ -46,17 +46,16 @@ rec {
       cd $APPDIR
       exec ./AppRun "$@"
     '';
-  });
+  } // (builtins.removeAttrs args [ "name" "src" "extraPkgs" ]));
```
Comment (tilpner) [inline pkgs/build-support/appimage/default.nix]: Would `with builtins; attrNames (functionArgs wrapAppImage)` be preferable here?

Comment (dtzWill) [reply pkgs/build-support/appimage/default.nix]: Oh, neat! I think so, no need to maintain two lists and perhaps is clearer re:intent.

**Comment by dtzWill** [comment]
* rebased onto master for final appimagetools that was merged
* switched to `functionArgs`, seems okay but take a look please
* I previously dropped `inherit name extraPkgs` in `wrapType1`  which I've restored, not sure what that's about.

re:third point maybe that isn't needed since `args` is merged in anyway, shrug.

**Comment by dtzWill** [comment]
(added commit removing the inherit in both variants, FWIW)

**Comment by dtzWill** [comment]
@GrahamcOfBorg eval

**Comment by dtzWill** [comment]
Look okay re:appimagetools changes, @tilpner ? :)

**Comment by carlosdagos** [comment]
Thanks for adding this one @dtzWill. Really looking forward to it getting merged :)

**Comment by tilpner** [comment]
Oh, sorry, I forgot about this one. Yes, looks fine!

**Comment by dtzWill** [comment]
It's up to 1.4.12 fwiw, I have that commit ready but can submit in a follow-up :).  But it's just a bump and hashfix happily.

**Comment by dtzWill** [comment]
@GrahamcOfBorg eval

**Comment by tilpner** [comment]
I had one remaining suspicion about at which levels `extraPkgs` is optional now. As it is now, this breaks #54696 with `called without required argument 'extraPkgs`.

**Comment by dtzWill** [comment]
Locally merging the linked standardnotes PR results in a tree that evaluates `standardnotes` without that error, AFAICT.  I think maybe https://github.com/NixOS/nixpkgs/pull/56259/commits/58ce19e097a9bc099bf86b45ca9317fddb59ccd0 fixed that, if unintentionally :).  Can you confirm, or have any ideas why I can't reproduce? Thanks!

**Comment by tilpner** [comment]
I'm not sure. I checked out standardnotes-appimagetools and picked all Appimage-related commits (`g cherry-pick 6dcdf27 056d827 58ce19e`).

I think 58ce19e097a9bc099bf86b45ca9317fddb59ccd0 is the one breaking it, not fixing it.

**Comment by dtzWill** [comment]
@GrahamcOfBorg build minetime standardnotes joplin-desktop

**Comment by dtzWill** [comment]
Well regardless rebase + some new changes and all is well re:eval.

\o/

Look good, @tilpner ? :)

**Comment by dtzWill** [comment]
> I think [58ce19e](https://github.com/NixOS/nixpkgs/commit/58ce19e097a9bc099bf86b45ca9317fddb59ccd0) is the one breaking it, not fixing it.

Sorry about this, you're right.  See comments on #54696; will probably PR if don't see that go through soon.

**Pull request merged**